pre-commit

```diff
@@ -18,33 +16,69 @@ declare -a descriptions=(
     "AWS account number"
     "16 byte hex (e.g. S3 bucket name)"
     "NINO"
-    "IP address"
     "Private key (e.g. rsa private key, openssh private key)"
-    "Email addresses"
     "Regions embedded as part of resource descriptions"
 )
 
+declare -a email_patterns=(
+    "\b[A-Za-z0-9._%+-]{1,}@[A-Za-z0-9]{1,}.[A-Za-z0-9]{1,4}.[A-Za-z0-9]{1,4}\b"
+)
+
+declare -a email_descriptions=(
+    "Email addresses"
+)
+
+declare -a ip_patterns=(
+    "\b(?:[0-9]{1,3}\.){3}[0-9]{1,3}\b"
+)
+
+declare -a ip_descriptions=(
+    "IP address"
+)
+
 if [ -d ".gitsecret" ]; then
     git secret hide
 fi
 
+
 match=0
 for i in "${!patterns[@]}"
 do
     git diff-index -p -M --cached HEAD -- | grep -v "Subproject commit" |
     grep '^+[^+]' | grep -Eq "${patterns[$i]}" &&
     echo "Blocking commit: ${descriptions[$i]} detected in patch" &&
+    echo "This part of your change triggered this issue:" &&
+    git diff-index -p -M --cached HEAD -- | grep -v "Subproject commit" |
+    grep '^+[^+]' | grep -E "${patterns[$i]}" &&
     ((match++))
 done
 
-if (( match > 0 )); then
+email_match=0
+for i in "${!email_patterns[@]}"
+do
+    git diff-index -p -M --cached HEAD -- | grep -v "snyk/snyk@"| grep -v "create-release@latest"| grep -v "checkout@master"| grep -v "Publish-Docker-Github-Action@master"| grep -v "docker@master"|
+    grep '^+[^+]' | grep -Eq "${email_patterns[$i]}" &&
+    echo "Blocking commit: ${email_descriptions[$i]} detected in patch" &&
+    echo "This part of your change triggered this issue:" &&
+    git diff-index -p -M --cached HEAD -- | grep -v "snyk/snyk@"| grep -v "create-release@latest"| grep -v "checkout@master"| grep -v "Publish-Docker-Github-Action@master"| grep -v "docker@master"|
+    grep '^+[^+]' | grep -E "${email_patterns[$i]}" &&
+    ((email_match++))
+done
+
+ip_match=0
+for i in "${!ip_patterns[@]}"
+do
+    git diff-index -p -M --cached HEAD -- | grep -v "169.254.169.254" | grep -v "0.0.0.0" | grep -v "127.0.0.1" |
+    grep '^+[^+]' | grep -Eq "${ip_patterns[$i]}" &&
+    echo "Blocking commit: ${ip_descriptions[$i]} detected in patch" &&
+    echo "This part of your change triggered this issue:" &&
+    git diff-index -p -M --cached HEAD -- | grep -v "169.254.169.254" | grep -v "0.0.0.0" | grep -v "127.0.0.1" |
```

Comment:
Ignoring localhost and local loopback addresses is fine. I wouldn't want to ignore 0.0.0.0 addresses though. Yes, they're not sensitive but it's a potentially massive security hole being introduced so should be flagged IMO.

Reply:
But that is what a review is for, not a pre-commit hook designed to catch sensitive data?

Reply:
I'd agree with @ChrisScottThomas here. The purpose of the hooks it to support the open source policy. Security concerns like opening up SG rules too much should be picked up during a PR, possibly by Terraform policy tests, but it's not something that needs to be caught before pushing to GitHub

Reply:
Fair enough. *I'd* prefer to be warned about such things as early as possible but I take your points gents. I guess it also enables us to put more lenient rules in whilst we work out how to do lock them down in due course.